### DISCUSSION TITLE
Running kimai2 on Apache2 Reverse Proxy with subdirectory

### DISCUSSION THREAD

**Discussion by Mnemonic78** [kimai/kimai#3044] [On-Premise: Installation & Updates]
Hello,
I'm a bit stuck configuring Kimai2 behind an Apache2 ReverseProxy. Since I have several services running I use subdirectories for each server. With my Nextcloud this works fine, only for Kimai I can't find a working configuration.
...
              <location /kimai2>
                    ProxyPass "http://192.168.178.88/"
                    ProxyPassReverse "http://192.168.178.88/"
                    ProxyHTMLEnable On
                </location>
                SSLProtocol -ALL +TLSv1.2
                SSLCipherSuite EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH
                SSLHonorCipherOrder on
                SSLCompression off

                ProxyPreserveHost On
                RequestHeader set X-Forwarded-Proto: "https"

                # If the proxy listens on other port than 443
                # RequestHeader set X-Forwarded-Port: <your-custom-port>

                Header always set Strict-Transport-Security "max-age=63072000;"
                Header set X-Content-Type-Options "nosniff".
                Header always append X-Frame-Options "SAMEORIGIN
                Header set Cache-Control "no-cache, no-store, no-transform
                Header set Pragma "no-cache
                Header set X-XSS-Protection "1; mode=block".
                Header set Referrer-Policy: "strict-origin-when-cross-origin".
                Header set Content-Security-Policy: "default-src https: data: 'self' 'unsafe-inline' 'unsafe-eval';"
                Header set feature-policy: "fullscreen 'self'"
                Header set x-permitted-cross-domain-policies "none".
...

I made the changes to the local.yaml too, still it doesn't want to work. Have also something to come further with the header forwarding. Unfortunately without success. Can someone help me further what I am missing? Locally the instance is accessible without problems.

**Comment by kevinpapst** [comment]
Generally using Kimai in a subdirectory is not supported. But maybe you are lucky if you follow the documentation:
https://www.kimai.org/documentation/webserver-configuration.html#reverse-proxy

**Reply by Mnemonic78** [reply]
I try this, but this description was for a nginx proxy and not for an apache. I try to adapt this, but it does not work. Using kimai in a subdirectory without reverse proxy on my old server was working.
Because this it must be the proxy configuration for apache.

**Reply by kevinpapst** [reply]
I haven't used Apache in years, so I am not a big help here.
Did you add the .env setting: https://github.com/kevinpapst/kimai2/blob/master/.env.dist#L15

Entirely general feedback:
1. Why do you add 100 different headers if it is not yet working? Start from a minimal setup without any additional stuff.
2. You did not even post what is not working.

**Reply by Mnemonic78** [reply]
1. I start with a minimal setup. This setup was working great for my nexcloud server behind the reverse proxy. I add a lot of because side was not loading. If I try to reach my kimai2 be https://MYSERVER/kimai2/ I got only https://MYSERVER/kimai2index.php/
2. I do the .env setting and I do the local.yaml setting.

**Reply by Mnemonic78** [reply]
I revert all header handling an start from zero but always the same result as described in 1. I hoped that some one here has a similar setup, because apache is a very common webserver.

**Reply by buoncri** [reply]
I am not shure that can help, but for services on the same machine can try to map subdirs like https://blabla.mydomain.com/kimai2 to https://kimai2.mydomain.com.
